app/code/Magento/ImportExport/Model/Import/Source/FileParser/ZipParserFactory.php

```diff
@@ -0,0 +1,118 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+
+namespace Magento\ImportExport\Model\Import\Source\FileParser;
+
+use Magento\Framework\App\Filesystem\DirectoryList;
+use Magento\Framework\Filesystem;
+
+class ZipParserFactory implements ParserFactoryInterface
+{
+    private $filesystem;
+    private $isZipAvailable;
+    private $parserFactory;
+
+    public function __construct(
+        Filesystem $filesystem,
+        ParserFactoryInterface $parserFactory,
+        $isZipAvailable = null
+    ) {
+        $this->filesystem = $filesystem;
+        $this->parserFactory = $parserFactory;
+        $this->isZipAvailable = $isZipAvailable ?? extension_loaded('zip');
+    }
+
+    public function create($path, array $options = [])
+    {
+        $this->assertZipFileIsParsable($path);
+
+        $sourceDirectory = $this->filesystem->getDirectoryRead(DirectoryList::ROOT);
+        $writeDirectory = $this->filesystem->getDirectoryWrite(DirectoryList::TMP);
+
+        $path = $sourceDirectory->getRelativePath($path);
+        $this->assertFileExistance($path, $sourceDirectory);
+
+        $zip = new \ZipArchive();
+
+        if ($zip->open($sourceDirectory->getAbsolutePath($path)) !== true) {
+            throw new CorruptedFileException($path);
+        }
+
+        $files = $this->fetchFileNamesFromZipFile($zip);
+
+        foreach ($files as $file) {
```

Comment:
I mean, what if the archive contains two or more valid files (products_1.csv, products_2.csv and so on).
Like before compression merchant split the original file into several files and after that compressed them all together.
Why do we choose first random file which fits for the parser, like  products_2.csv
but not all?